sympy/combinatorics/tests/test_fp_groups.py

```diff
@@ -171,10 +171,21 @@ def test_fp_subgroup():
     f = FpGroup(F, [x**4, y**2, x*y*x**-1*y])
     S = FpSubgroup(f, [x*y])
     assert (x*y)**-3 in S
+    K, T = f.subgroup([x*y], homomorphism=True)
+    assert T(K.generators) == [y*x**-1]
+    assert T.is_injective()
```

Comment:
You should check that `T.image().order() = K.order()` to verify the image is indeed isomorphic to the subgroup and possibly `T(K.generators) in S` would be a better check for generators. I would actually write a unit test as a subfunction here to avoid repetition. Say:
```
def _test(K, T):
   assert T(K.generators) in S
   assert T.is_injective()
   assert T.image.order() == S.order()
```
and just run `_test(f.subgroup([x**-1*y*x], homomorphism=True)`, etc after that.